### PR TITLE
SSCS-10978: initial implementation of sending generic letter

### PR DESCRIPTION
**Pull request by AndreiTrushkov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10978

### Change description ###
initial implementation of sending generic letter

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java [hunk 1]
```diff
@@ -518,6 +605,38 @@ private void verifyNotificationIsSent(NotificationHandler.SendNotification sende
...
+        var wrapper = buildBaseWrapper(appellant, eventType, representative, Benefit.PIP, "Online", READY_TO_LIST.getId());
+
+        if (documentsSelection != null) {
+            wrapper.getNewSscsCaseData().setDocumentSelection(documentsSelection);
+
+            var sscsLink = DocumentLink.builder().documentUrl("sscs_url").documentFilename("sscs").build();
+            var dwpLink = DocumentLink.builder().documentUrl("dwp_url").documentFilename("dwp").build();
+
+            if (sscsDocument) {
+                wrapper.getNewSscsCaseData().getSscsDocument().add(SscsDocument.builder()
```
Comment (jarekPierchala) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java:619]: `wrapper.getNewSscsCaseData()` to separate var

File: src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java [hunk 2]
```diff
@@ -518,6 +605,38 @@ private void verifyNotificationIsSent(NotificationHandler.SendNotification sende
...
+                        .value(SscsDocumentDetails.builder().documentLink(sscsLink).documentFileName("sscsDocument").build())
```
Comment (jarekPierchala) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java:620]: `SscsDocumentDetails.builder().documentLink(sscsLink).documentFileName("sscsDocument").build()`
to separate var and reference the var

File: src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java [hunk 3]
```diff
@@ -518,6 +605,38 @@ private void verifyNotificationIsSent(NotificationHandler.SendNotification sende
...
+                        .build());
+            } else {
+                wrapper.getNewSscsCaseData().getSscsDocument().clear();
+            }
+
+            if (dwpDocument) {
+                wrapper.getNewSscsCaseData().setDwpDocuments(List.of(DwpDocument.builder()
+                        .value(DwpDocumentDetails.builder().documentLink(dwpLink).documentFileName("dwpDocument").build())
```
Comment (jarekPierchala) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java:628]: `DwpDocumentDetails.builder().documentLink(dwpLink).documentFileName("dwpDocument").build()` separate var

**Review by jarekPierchala — APPROVED** [review]

**Review by hajam09 — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Comment by AndreiTrushkov** [comment]
not needed

**Pull request closed without merging**